backend/src/config/commonFunctions.ts

```diff
@@ -0,0 +1,51 @@
+/*
+This file contains functions that are shared across the app
+ */
+
+/* istanbul ignore next */
+/**
+ * Returns the string that returns our NodeDescriptor object
+ * @param node Name of the variable in the query
+ */
+export function neo4jReturnNodeDescriptor(node: string): string {
+  return `ID(${node}) as id`;
+}
+
+/* istanbul ignore next */
+/**
+ * Returns the string that returns our EdgeDescriptor object
+ * @param edge Name of the edge in the query
+ * @param from Name of the start-edge in the query
+ * @param to Name of the end-edge in the query
+ */
+export function neo4jReturnEdgeDescriptor(
+  edge: string,
+  from: string,
+  to: string
+): string {
+  return `ID(${edge}) as id, ID(${from}) as from, ID(${to}) as to`;
+}
+
+/* istanbul ignore next */
+/**
+ * Returns the string that returns our Node object
+ * @param node Name of the variable in the query
+ */
+export function neo4jReturnNode(node: string): string {
+  return `ID(${node}) as id, labels(${node}) as labels, properties(${node}) as properties`;
```

Comment:
This code allows database injection attacks!

Reply:
[see previous comment](https://github.com/amosproj/amos-ss2021-project2-context-map/pull/115#discussion_r632931809)

Reply:
So are the parameter values known?  
Then add them in the function definition, instead of receiving arguments of type `string` and add a runtime check.

If the values are not known up-front, add a warning to the function comment, that the function MUST NOT be called with non-constant string arguments, due to possible database injection attacks.

The thing here is that currently, there is no attack possible, but if further development  in the backend does not use these functions with care, attacks may get possible then.